.github/workflows/cloud-deploy-reusable-docker.yml

```diff
@@ -0,0 +1,204 @@
+        required: false
+        default: false
+        type: boolean
+      # The tag under which the image is built, it should align with the name in the project command docker:build
+      local_tag:
+        required: true
+        type: string
+      # The environment tag. Possible values are dev, stg, and prod. This is not a required input of type string.
+      env_tag:
+        required: false
+        type: string
+      enterprise:
+        required: false
+        type: boolean
+    outputs:
+      docker_image:
+        description: 'The image that was built'
+        value: ${{ jobs.reusable_docker.outputs.docker_image }}
+
+jobs:
+  job-setup:
+    runs-on: ubuntu-latest
+    outputs:
+      platforms: ${{ steps.set-platforms.outputs.platforms }}
+      dockerName: ${{ steps.set-docker-name.outputs.dockerName }}
+    steps:
+      - id: set-docker-name
+        run: |
+          if [[ "${{ inputs.enterprise }}" == "true" ]]; then
+            echo "dockerName=${{ inputs.package_name }}-ee" >> "$GITHUB_OUTPUT"
+          else
+            echo "dockerName=${{ inputs.package_name }}" >> "$GITHUB_OUTPUT"
+          fi
+
+      - id: set-platforms
+        run: |
+          if [[ "${{ inputs.environment }}" == "Production" ]]; then
+            echo "platforms=linux/amd64" >> "$GITHUB_OUTPUT"
+          else
+            echo "platforms=linux/amd64" >> "$GITHUB_OUTPUT"
+          fi
+
+  reusable_docker:
+    needs: [job-setup]
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    outputs:
+      docker_image: ${{ steps.save-image-to-output.outputs.IMAGE }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: ${{ inputs.enterprise }}
+          token: ${{ secrets.SUBMODULES_TOKEN }}
+
+      - name: Prepare
+        id: prepare
+        shell: bash
+        run: |
+          service=${{ inputs.package_name }}
+          servicename=${service#*/} 
+          echo "SERVICE_NAME=$servicename" >> "$GITHUB_OUTPUT"
+
+      - uses: ./.github/actions/setup-project
+        with:
+          slim: 'true'
+          submodules: ${{ inputs.enterprise }}
+
+      - uses: crazy-max/ghaction-setup-docker@v2
+        with:
+          version: v24.0.6
+          daemon-config: |
+            {
+              "features": {
+                "containerd-snapshotter": true
+              }
+            }
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: ${{ needs.job-setup.outputs.platforms }}
+
+      - name: Set Up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: 'image=moby/buildkit:v0.13.1'
+
+      - name: Build, tag, and push image to ghcr.io
+        id: build-image
+        if: ${{ inputs.env_tag == 'dev' || inputs.env_tag == 'stg' }}
```

Comment:
If we pass environment as an option, building should be per environment so there shouldn't be any need for conditionals here.